### PR TITLE
ignore escaped quotes in strings, unescape double slashes

### PR DESCRIPTION
**Pull request by tymofij**
Fixes https://github.com/paveldedik/neon-py/issues/9

### CODE REVIEW

**Comment by paveldedik** [comment]
@tymofij The same as the other MR, can you please rebase from master and run pre-commit over this branch?

**Comment by tymofij** [comment]
Merged master and applied black formatting, the CI should pass now.

**Comment by paveldedik** [comment]
@tymofij Seems like py2.7 tests are failing. Do you think the fix would be too hard to do? I'm thinking about dropping support for python 2.7, so if it's more like 5 min to fix it, I'll just drop support for py2.7. Let me know.

File: tests/test_types.py [hunk 1]
```diff
@@ -60,3 +60,10 @@ def test_datetime():
         datetime(2015, 5, 10),
     ]
     assert neon.decode(NEON_DATETIME) == expected
+
+
+def test_string_escaping():
+    assert neon.decode('key: "msg"') == {"key": "msg"}
+    assert neon.decode(r'key: "msg \" end"') == {"key": 'msg " end'}
+    assert neon.decode(r"key: 'msg \' end'") == {"key": "msg ' end"}
+    assert neon.decode(r'src: "\\usr\\share"') == {"src": r"\usr\share"}
```
Comment (paveldedik) [inline tests/test_types.py:69]:
```suggestion
    assert neon.decode('src: "\\\\usr\\\\share"') == {"src": "\\usr\\share"}
```

File: tests/test_types.py [hunk 2]
```diff
@@ -60,3 +60,10 @@ def test_datetime():
         datetime(2015, 5, 10),
     ]
     assert neon.decode(NEON_DATETIME) == expected
+
+
+def test_string_escaping():
+    assert neon.decode('key: "msg"') == {"key": "msg"}
+    assert neon.decode(r'key: "msg \" end"') == {"key": 'msg " end'}
+    assert neon.decode(r"key: 'msg \' end'") == {"key": "msg ' end"}
```
Comment (paveldedik) [inline tests/test_types.py:68]:
```suggestion
    assert neon.decode("key: 'msg \\' end'") == {"key": "msg ' end"}
```

File: tests/test_types.py [hunk 3]
```diff
@@ -60,3 +60,10 @@ def test_datetime():
         datetime(2015, 5, 10),
     ]
     assert neon.decode(NEON_DATETIME) == expected
+
+
+def test_string_escaping():
+    assert neon.decode('key: "msg"') == {"key": "msg"}
+    assert neon.decode(r'key: "msg \" end"') == {"key": 'msg " end'}
```
Comment (paveldedik) [inline tests/test_types.py:67]:
```suggestion
    assert neon.decode('key: "msg \\" end"') == {"key": 'msg " end'}
```

**Comment by paveldedik** [comment]
> @tymofij Seems like py2.7 tests are failing. Do you think the fix would be too hard to do? I'm thinking about dropping support for python 2.7, so if it's more like 5 min to fix it, I'll just drop support for py2.7. Let me know.

Nevermind, I found a solution.

**Pull request merged**